client/src/main/java/gui/playing/PlayBoardController.java

```diff
@@ -20,34 +29,123 @@
 public class PlayBoardController implements Initializable {
 
   private final Client client;
-  private final MessageProcessor reactor;
+  private List<Position> positions;
+  private final MessageProcessor processor;
+  private SeaField lastField;
 
   @FXML
   private GridPane shipBoard;
+  @FXML
+  private GridPane targetBoard;
+  @FXML
+  private TextField winning;
 
-  public PlayBoardController(Client client, MessageProcessor reactor) {
+  public PlayBoardController(Client client, MessageProcessor processor, List<Position> positions) {
     this.client = client;
-    this.reactor = reactor;
+    this.processor = processor;
+    this.positions = positions;
   }
 
   @Override
   public void initialize(URL location, ResourceBundle resources) {
     populateSeaWithSeaFields();
+    shipBoard.setDisable(true);
+    winning.setText("wait");
+    populateOpponentBoardWithFleet();
+
+    winning.addEventHandler(UpdateWhenHitEvent.UPDATE, updateBoardWhenHit);
+    winning.addEventHandler(UpdateWhenMissedEvent.UPDATE_MISSED, updateBoardWhenMissed);
+    winning.addEventHandler(YourTurnEvent.TURN, enableBoard);
+    winning.addEventHandler(YouMissedEvent.WAIT, youMissed);
+    winning.addEventHandler(YouWinEvent.WIN, youWin);
+    winning.addEventHandler(YouLostEvent.LOST, youLost);
+    winning.addEventHandler(YouHitEvent.HIT, youHit);
+
+    new Thread(() -> {
+      Boolean isRunning = true;
+      while (isRunning) {
+        String message = client.getMessage();
+        processor.processMessage(message);
+        if (message.equals("WIN") || message.equals("LOST")) {
+          isRunning = false;
+        }
+      }
+    }).start();
   }
 
   private void populateSeaWithSeaFields() {
     for (int i = 0; i < 10; i++) {
       for (int n = 0; n < 10; n++) {
-        SeaField field = new SeaField(i, n);
+        SeaField field = new SeaField(i, n, FieldSize.BIG);
         field.addEventHandler(MouseEvent.MOUSE_CLICKED, event -> {
           client.sendMessage(field);
           field.marked();
-          reactor.processMessage(field, client.getMessage());
+          lastField = field;
         });
         shipBoard.getChildren().add(field);
         GridPane.setConstraints(field, i, n);
       }
     }
   }
+
+  private void populateOpponentBoardWithFleet() {
+    for (Position position : positions) {
+      Mast smallMast = new Mast(position.getRow(), position.getColumn(), FieldSize.SMALL);
+      targetBoard.getChildren().add(smallMast);
+      GridPane.setConstraints(smallMast, position.getColumn(), position.getRow());
+    }
+  }
+
+  private void putFieldOnOpponentBoard(Field field) {
+    Platform.runLater(() -> {
+      field.markedAsHit();
+      targetBoard.getChildren().add((Node) field);
+      GridPane.setConstraints((Node) field, field.getRow(), field.getColumn());
```

Comment:
double casting without instanceof.
you could do
Node node = (Node) filed;
and reuse it instead double casting.